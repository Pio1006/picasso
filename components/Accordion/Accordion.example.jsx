/* eslint-disable react/no-multi-comp, react/require-optimization, react/prop-types */
import React from 'react'
import { storiesOf } from '@storybook/react'

import StoryTeller from '../../.storybook/StoryTeller'
import Accordion from './'
import Spacer from '../Spacer'
import Button from '../Button'

const stories = storiesOf('Accordion', module)

class ExpandedControl extends React.Component {
  constructor () {
    super()

    this.state = {
      expanded: false
    }

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    const { expanded } = this.state

    this.setState({
      expanded: !expanded
    })
  }

  render () {
    const { children } = this.props
    const { expanded } = this.state

    return (
      <div style={{ display: 'flex', flexDirection: 'column' }}>
        <Spacer bottom={2}>
          <Button onClick={this.handleClick}>Expand/Collapse</Button>
        </Spacer>

        <div>{React.cloneElement(children, { expanded })}</div>
      </div>
    )
  }
}

const SummaryDogDefinitionPanel = () => <div>What is a dog?</div>
const DetailsDogDefinitionPanel = () => (
  <div>
    A dog is a type of domesticated animal. Known for its loyalty and
    faithfulness, it can be found as a welcome guest in many households across
    the world.
  </div>
)

// Chapter Accordion

const teller = new StoryTeller(
  'Accordion',
  'Accordions store information behind collapsible sections, allowing for more information to be stored in a limited amount of space.'
)
const chapter = teller.addChapter()

chapter.addSection(
  'Default',
  'Styled sections is a default behaviour of Accordion when `expanded` prop is not specified (uncontrolled).',
  () => (
    <div style={{ width: '430px' }}>
      <Accordion
        Details={<DetailsDogDefinitionPanel />}
        Summary={<SummaryDogDefinitionPanel />}
      />
    </div>
  )
)

chapter.addSection(
  'Controlled',
  'You can control of expansion or collapsing of the Details panel by passing `expanded` prop.',
  () => (
    <ExpandedControl>
      <Accordion Details={<DetailsDogDefinitionPanel />} expanded={false} />
    </ExpandedControl>
  )
)
stories.addWithChapters('Accordion', teller.toStory())

// Chapter Accordion Group

const tellerGroup = new StoryTeller(
  'Group',
  'Default Accordions with styled sections in a group.'
)
const chapterGroup = tellerGroup.addChapter()

const SummaryDogKindPanel = () => <div>What kinds of dogs are there?</div>
const DetailsDogKindPanel = () => (
  <div>
    There are many breeds of dogs. Each breed varies in size and temperament.
    Owners often select a breed of dog that they find to be compatible with
    their own lifestyle and desires from a companion.
  </div>
)
const SummaryDogAcquirePanel = () => <div>How do you acquire a dog?</div>
const DetailsDogAcquirePanel = () => (
  <div>
    Three common ways for a prospective owner to acquire a dog is from pet
    shops, private owners, or shelters. A pet shop may be the most convenient
    way to buy a dog. Buying a dog from a private owner allows you to assess the
    pedigree and upbringing of your dog before choosing to take it home. Lastly,
    finding your dog from a shelter, helps give a good home to a dog who may not
    find one so readily.
  </div>
)

chapterGroup.addSection('Default', null, () => (
  <div style={{ width: '430px' }}>
    <Accordion
      Details={<DetailsDogDefinitionPanel />}
      Summary={<SummaryDogDefinitionPanel />}
    />
    <Accordion
      Details={<DetailsDogKindPanel />}
      Summary={<SummaryDogKindPanel />}
    />
    <Accordion
      Details={<DetailsDogAcquirePanel />}
      Summary={<SummaryDogAcquirePanel />}
    />
  </div>
))

stories.addWithChapters('Accordion Group', tellerGroup.toStory())
