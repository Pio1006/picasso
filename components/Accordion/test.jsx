/* eslint-disable react/no-multi-comp */
import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import Accordion from './index'

const renderAccordion = (props = {}) => {
  return render(<Accordion {...props} />)
}

afterEach(cleanup)

const Summary = () => <div>Fryderyk Chopin</div>

const Details = () => (
  <div>
    Fryderyk Chopin was born in Å»elazowa Wola, 46 kilometres west of Warsaw, in
    what was then the Duchy of Warsaw, a Polish state established by Napoleon.
    The parish baptismal record gives his birthday as 22 February 1810, and
    cites his given names in the Latin form Fridericus Franciscus (in Polish, he
    was Fryderyk Franciszek). However, the composer and his family used the
    birthdate 1 March, which is now generally accepted as the correct date.
  </div>
)

describe('default version for sections', () => {
  let api

  beforeEach(() => {
    api = renderAccordion({
      Details: <Details />,
      Summary: <Summary />
    })
  })

  test('should render default version', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })

  test('should render expanded version after click on summary', () => {
    const { container, getByTestId } = api
    const summary = getByTestId('panel-summary')

    fireEvent.click(summary)

    expect(container).toMatchSnapshot()
  })
})

describe('controlled version', () => {
  test('should render expanded version', () => {
    const { container } = renderAccordion({
      Details: <Details />,
      expanded: true
    })

    expect(container).toMatchSnapshot()
  })

  test('should render collapsed version', () => {
    const { container } = renderAccordion({
      Details: <Details />,
      expanded: false
    })

    expect(container).toMatchSnapshot()
  })
})
