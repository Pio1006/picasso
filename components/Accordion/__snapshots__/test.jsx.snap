// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controlled version should render collapsed version 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root"
  >
    <div
      aria-hidden="true"
      class="MuiCollapse-container"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            class="MuiExpansionPanelDetails-root ExpansionPanelDetails-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`controlled version should render expanded version 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-expanded"
  >
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            class="MuiExpansionPanelDetails-root ExpansionPanelDetails-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`default version for sections should render default version 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root MuiExpansionPanelSummary-root WithStyles-ExpansionPanelSummary--root Accordion-defaultSummary"
      data-testid="panel-summary"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content"
      />
      <div
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root Accordion-expandIcon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiCollapse-container"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            class="MuiExpansionPanelDetails-root ExpansionPanelDetails-root Accordion-defaultDetails"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`default version for sections should render expanded version after click on summary 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-expanded"
  >
    <div
      aria-expanded="true"
      class="MuiButtonBase-root MuiExpansionPanelSummary-root WithStyles-ExpansionPanelSummary--root Accordion-defaultSummary MuiExpansionPanelSummary-expanded"
      data-testid="panel-summary"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content MuiExpansionPanelSummary-expanded"
      />
      <div
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiExpansionPanelSummary-expanded"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root Accordion-expandIcon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container"
      style="min-height: 0px; height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            class="MuiExpansionPanelDetails-root ExpansionPanelDetails-root Accordion-defaultDetails"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
